File: BOJ/2884.py
a, b=map(int, input().split())
t = (60 * a) + b - 45
h = int(t // 60)
m = int(t % 60)
if h > 23:
    h = 0
elif h < 0:
    h = 23
if m < 0:
    m = 60 + m
if m > 59:
    h = h+ 1
print(h,m)

File: BOJ/2884.js
const fs = require('fs');
const input = fs.readFileSync('/dev/stdin').toString().split(' ');
a = parseInt(input[0]);
b = parseInt(input[1]);

if (b > 45) {
    b -= 45;
}
else {
    b += 15;
    a--;
    if (a < 0) {
        a = 23;
    }
}
console.log(a, b);
